exporter/googlecloudexporter/exporter.go

```diff
@@ -150,3 +160,39 @@ func (e *exporter) Shutdown(ctx context.Context) error {
 
 	return nil
 }
+
+// appendMetricAttrs appends metric attributes if not present
+func (e *exporter) appendMetricAttrs(md *pmetric.Metrics) {
+	for i := 0; i < md.ResourceMetrics().Len(); i++ {
+		resourceAttrs := md.ResourceMetrics().At(i).Resource().Attributes()
+		_, hostNameExists := resourceAttrs.Get(string(semconv.HostNameKey))
+		_, hostIDExists := resourceAttrs.Get(string(semconv.HostIDKey))
+		if !hostNameExists && !hostIDExists && e.appendHost {
+			resourceAttrs.InsertString(string(semconv.HostNameKey), hostname)
+		}
+	}
+}
+
+// appendLogAttrs appends log attributes if not present
+func (e *exporter) appendLogAttrs(ld *plog.Logs) {
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		resourceAttrs := ld.ResourceLogs().At(i).Resource().Attributes()
+		_, hostNameExists := resourceAttrs.Get(string(semconv.HostNameKey))
+		_, hostIDExists := resourceAttrs.Get(string(semconv.HostIDKey))
+		if !hostNameExists && !hostIDExists && e.appendHost {
+			resourceAttrs.InsertString(string(semconv.HostNameKey), hostname)
+		}
+	}
```

Comment:
if e.appendHost is false, I think we should bail out before the loop, since we won't ever add the attribute.